### PR TITLE
Add new `Style/EmptyLinesAroundBeginBody` cop

### PR DESCRIPTION
**Pull request by pocke**
Example
------

```ruby
 # good
begin
  foo
end

 # bad
begin

  foo

end
```

Note
--------

This cop does not detect the following code.

```ruby
begin
  foo
  # Empty line is here!

rescue
  bar
end

begin
  foo
rescue

  # Empty line is here!
  bar
end

begin
  foo
  # Empty line is here!

ensure
  bar
end
```

There are two reasons for this.

First, the feature can't be implemented by `check` method that is defined by `EmptyLinesAroundBody` mixin.

Second, I think the feature should be implemented as another cop(e.g. `Style/EmptyLinesAroundBeginSection`).
I would like to check empty lines around begin-end block body always.
However, I add an empty line around `ensure`, `rescue`, `else` keywords sometimes.

```ruby
begin
  foo
  bar
  baz

rescue
  something
end
```

So, the features should be separateness to disable the one only.

I'll implement the cop for empty lines around the keywords with another PR.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: config/enabled.yml [hunk 1]
```diff
@@ -223,6 +223,10 @@ Style/EmptyLinesAroundAccessModifier:
   Description: "Keep blank lines around access modifiers."
   Enabled: true
 
+Style/EmptyLinesAroundBeginBody:
```
Comment (bbatsov) [inline config/enabled.yml]: Don't we have a style guide rule for this?

File: lib/rubocop/cop/style/empty_lines_around_begin_body.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      #
+      #   begin
+      #
+      #     ...
+      #
+      #   end
+      class EmptyLinesAroundBeginBody < Cop
+        include EmptyLinesAroundBody
+
+        KIND = 'begin'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_lines_around_begin_body.rb]: This looks a bit confusing. Maybe we should turn it into a method instead?

File: spec/rubocop/cop/style/empty_lines_around_begin_body_spec.rb [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+require 'spec_helper'
+
+describe RuboCop::Cop::Style::EmptyLinesAroundBeginBody do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  shared_examples :offense do |name, code, correction|
+    it "registers an offense for #{name} with a blank" do
+      inspect_source(cop, code)
+      expect(cop.offenses.size).to eq(1)
```
Comment (bbatsov) [inline spec/rubocop/cop/style/empty_lines_around_begin_body_spec.rb]: I don't see a check for the message generated by the cop.

File: spec/rubocop/cop/style/empty_lines_around_begin_body_spec.rb [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+  shared_examples :accepts do |name, code|
+    it "accepts #{name}" do
+      inspect_source(cop, code)
+      expect(cop.offenses).to be_empty
+    end
+  end
+
+  include_examples :offense, 'begin body starting', <<-CODE, <<-CORRECTION
+begin
```
Comment (bbatsov) [inline spec/rubocop/cop/style/empty_lines_around_begin_body_spec.rb]: Same remark as from your other PR - use `strip_margin` so we can pretty up the heredocs.

Comment (bbatsov) [inline spec/rubocop/cop/style/empty_lines_around_begin_body_spec.rb]: We should probably add some cop for newer Rubies and a rule in the style guide to use the new heredocs syntax - it's so much nicer...

File: config/enabled.yml (hunk 1 above)
Comment (pocke) [inline config/enabled.yml]: No, we don't have the style guide. And we also don't have style guides for other `EmptyLines*` cops.
I'll add the documentation.

File: lib/rubocop/cop/style/empty_lines_around_begin_body.rb (hunk 2 above)
Comment (pocke) [inline lib/rubocop/cop/style/empty_lines_around_begin_body.rb]: Why are you confusing? I think it is good to keep constant assignment.

Note:  The `begin` keyword should be surrounded by backtick(I'll fix it).

File: spec/rubocop/cop/style/empty_lines_around_begin_body_spec.rb (hunk 3 above)
Comment (pocke) [inline spec/rubocop/cop/style/empty_lines_around_begin_body_spec.rb]: I've made to check it 👍

**Pull request merged**